app.ts

```diff
@@ -0,0 +1,411 @@
+				$("#records-table-body").empty();
+				this.loader();
+				this.currentToID = toID;
+				this.currentFromID = fromID;
+				if (toID >= (count - 1)) {
+					this.currentToID = (count - 1);
+					this.currentFromID = this.currentToID - (maxRecords - 1);
+				} else if (this.currentPage === 1) {
+					this.currentFromID = 0;
+				}
+				stringCount = count.toLocaleString().replace(/,/g, " ");
+				return this.apiManager.getRecords(this.currentFromID, this.currentToID);
+			})
+			.then(records => {
+				$('.results').empty().append(`Displaying ID's ${this.currentFromID} - ${this.currentToID} out of ${stringCount}`);
+				for (const record of records) {
+					$("#records-table-body").append(`<tr class="body-row">`);
+					for (const value of record) {
+						let isSearchValue = value === inputNumber;
+						$(".body-row:last-child").append(
+							`<td>
+								<span class="${isSearchValue ? 'highlight' : ''}">${value}</span>
+							</td>`);
+					}
+					$("#records-table-body").append(`</tr>`);
+				}
+				this.isFunctionRunning = false;
+			})
+			.catch(error => {
+				alert("Something went wrong. Click on the button to refresh the page.");
+				location.reload();
+				this.isFunctionRunning = false;
+			});
+	}
+
+	/** Handles pagination functionality and rendering it on the DOM.*/
+	pageNumbers(start: number, end: number): Promise<void> {
+		this.paginationStart = start;
+		this.paginationEnd = end;
+		return this.apiManager.getRecordCount()
+			.then(count => {
+				$('.pagination').empty();
+				let maxRecords = this.recordsPerPage();
+				// This is the last page of all the records. It's calculated based on the amount of records showed
+				// on the table and the count of the records.
+				let lastPage = Math.ceil((count - 1) / maxRecords);
+				if (lastPage <= this.paginationEnd && lastPage >= this.paginationStart) {
+					this.paginationEnd = lastPage;
+					$(".next").css({ display: "none" });
+				} else {
+					$(".next").css({ display: "block" });
+				}
+				if (this.paginationStart < 1) {
+					this.paginationStart = 1;
+				}
+				for (let i = this.paginationStart; i <= this.paginationEnd; i++) {
+					let isActive = i == this.currentPage;
+					$(".pagination").append(
+						`<a id="page-${i}" value="${i}" class="pagination-item ${isActive ? 'active' : ''}">${i}</a>`
+					);
+				}
+				if (this.paginationStart === 1) {
+					$(".prev").css({ display: "none" });
+				} else {
+					$(".prev").css({ display: "block" });
+				}
+			})
+			.catch(error => {
+				throw new Error(`Failed rendering the pagination: ${error}`);
+			});
+	}
+
+	/** Handles all the functionality related to pagination. */
+	initializePagination(): void {
+		// Render the specific amount of records to the DOM based on the current page that it gets from the 
+		// element's attribute value. 
+		$(".pagination").on("click", ".pagination-item", (event) => {
+			$('.pagination-item').prop('disabled', true);
+			return this.apiManager.getRecordCount()
```

Comment:
You don't have to return here. Just run it.
```suggestion
			this.apiManager.getRecordCount()
```